core/src/dird/ua_status.cc

```diff
@@ -478,6 +478,24 @@ static bool show_scheduled_preview(UaContext*,
   return true;
 }
 
+std::string get_subscription_status_signature_source_text(UaContext* ua,
```

Comment:
you're only using `ua->db`, so maybe consider `BareosDb*` instead of `UaContext*` or maybe `BareosDb&`.